streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java

```diff
@@ -791,49 +751,36 @@ public void shouldWriteLatestOffsetsToCheckpointOnShutdown() throws Exception {
         final List<KeyValue<Long, Long>> expectedResult = computeExpectedResult(writtenData);
 
         try (final KafkaStreams streams = getKafkaStreams("streams", true, "appDir", 1, eosConfig, MAX_POLL_INTERVAL_MS)) {
+            writeInputData(writtenData);
 
             startKafkaStreamsAndWaitForRunningState(streams, MAX_WAIT_TIME_MS);
 
-            writeInputData(writtenData);
-
             waitForCondition(
                     () -> commitRequested.get() == 2, MAX_WAIT_TIME_MS,
                     "SteamsTasks did not request commit.");
 
-            final List<KeyValue<Long, Long>> committedRecords = readResult(writtenData.size(), CONSUMER_GROUP_ID);
+            final List<KeyValue<Long, Long>> committedRecords = readResult(SINGLE_PARTITION_OUTPUT_TOPIC, writtenData.size(), CONSUMER_GROUP_ID);
```

Comment:
This is inspired by https://github.com/apache/kafka/pull/11878: we produce first then start streams.